### DISCUSSION TITLE
Validating DNSSEC signatures with the stub resolver

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [NLnetLabs/domain#366] [Q&A] [answered]
I'm using the stub resolver to perform SRV-based discovery of services on a domain.

I'd like to ascertain whether the response has a valid DNSSEC signature. Do I understand correctly that the API for this is not yet finalised?

**Comment by Philip-NLnetLabs** [comment, marked as answer]
Yes, the API is not considered stable. It is not clear how big the changes will be. The validator itself needs to get a worker task at some point. But the validator also uses the client transport code which is also unstable.

Most likely future changes will at most require minor code changes, but we can't be sure.

**Reply by WhyNotHugo** [reply]
I have a follow-up on this, since I want to clearly document the limitations of [my tool](https://sr.ht/~whynothugo/vdirsyncer-rs/) which uses this crate.

Suppose that my local DNS server is a [validating resolver](https://www.rfc-editor.org/rfc/rfc4033#section-5) and it indicates that a DNSSEC signature is invalid for a given response (using RCODE=2, "Server Failure"; see [[RFC4035](https://www.rfc-editor.org/rfc/rfc4035)]).

In this case the stub resolver would return an `Err` rather than the invalid result, right?

**Reply by partim** [reply]
Correct.
